src/index.js

```diff
@@ -1,49 +1,20 @@
 import fs from 'fs'
 import path from 'path'
-import yaml from 'yamljs'
 
-import {
-  parse as parseFor,
-  usage as usageFor,
-} from 'subcli'
-
-function loadCommandDescriptor(commandConfigFilename) {
-  const {command} = yaml.load(commandConfigFilename)
-  return command
-}
-
-function genCommand(commandDescriptor) {
-  return {
-    commandDescriptor,
-    parse: argv => parseFor(commandDescriptor, argv),
-    usage: (args = null) => usageFor(commandDescriptor, args),
-  }
-}
-
-function filterOutInactiveSubcommandDescriptors(commandDescriptor) {
-  if (!commandDescriptor.commands) {
-    return commandDescriptor
-  }
-  const filteredSubcommands = filterOutInactiveCommandDescriptors(commandDescriptor.commands)
-  const filteredCommandDescriptor = Object.assign({}, commandDescriptor, {commands: filteredSubcommands})
-  return filteredCommandDescriptor
-}
-
-function filterOutInactiveCommandDescriptors(commandDescriptors) {
-  return commandDescriptors.map(commandDescriptor => filterOutInactiveSubcommandDescriptors(commandDescriptor))
-    .filter(commandDescriptor => {
-      return !commandDescriptor._inactive
-    })
+function filterOutInactiveCommands(commands) {
+  return commands.filter(command => {
+    return !command.commandDescriptor._inactive
+  })
 }
 
 function genCommandMap() {
-  const commandConfigDir = path.resolve(__dirname, '..', 'config', 'commands')
-  const commandConfigFilenames = fs.readdirSync(commandConfigDir)
-    .filter(filename => filename.match(/\.yaml$/))
-    .map(filename => path.resolve(commandConfigDir, filename))
-  const commandDescriptors = commandConfigFilenames.map(filename => loadCommandDescriptor(filename))
-  return filterOutInactiveCommandDescriptors(commandDescriptors)
-    .map(commandDescriptor => genCommand(commandDescriptor))
+  const commandSrcDir = path.resolve(__dirname, 'commands')
+  const commandSourceFilenames = fs.readdirSync(commandSrcDir)
+    .filter(filename => filename.match(/\.js$/))
+    .map(filename => path.resolve(commandSrcDir, filename))
+  const commands = commandSourceFilenames.map(filename => require(filename))
+  return filterOutInactiveCommands(commands)                  // ensure that it's not marked _inactive
+    .filter(command => typeof command.invoke === 'function')  // ensure that there's an implementation
```

Comment:
Mostly for consistency, since we still need to obey the flag for subcommands, and the "command descriptor" shape is the same whether it's a parent command or subcommand.